File: BFF-client/BFF-client.Api/Models/Credit/CreateCreditDto.cs
namespace BFF_client.Api.Models.Credit
{
    public class CreateCreditDto
    {
        public Guid? AccountId { get; set; } = null;

        public Guid BillId { get; set; }

        public long Principal { get; set; }

        public DateOnly CompletionDate { get; set; }

        public Guid CreditTermsId { get; set; }
    }
}

File: android-client/data/src/main/java/ru/hits/hubabank/data/network/credit/model/CreateCreditDto.kt
package ru.hits.hubabank.data.network.credit.model

import kotlinx.serialization.Serializable
import ru.hits.hubabank.domain.credit.model.CreateCredit

@Serializable
internal data class CreateCreditDto(
    val billId: String,
    val principal: Long,
    val completionDate: String,
    val creditTermsId: String,
)

internal fun CreateCredit.toDto(): CreateCreditDto {
    return CreateCreditDto(
        billId = billId,
        principal = principal,
        completionDate = completionDate.toString(),
        creditTermsId = creditTermsId
    )
}
